components/script/dom/htmlformelement.rs

```diff
@@ -1332,6 +1390,29 @@ impl<'a> FormSubmitter<'a> {
                 ),
         }
     }
+
+    // https://html.spec.whatwg.org/multipage/#concept-submit-button
+    fn is_submit_button(&self) -> bool {
```

Comment:
The spec reads to me as if only a button element could be potentially a submit button, if that is correct we should remove the `InputElement` branch.